src/modules/site-v2/base/views/tools/phenotype_database.py

```diff
@@ -59,12 +63,89 @@ def results_columns():
 @phenotype_database_bp.route('')
 @cache.memoize(60*60)
 def phenotype_database():
+  """
+    Phenotype Database table (non-bulk)
+  """
+  # Get the list of traits for non-bulk files
+  try:
+    traits_non_bulk = query_phenotype_metadata()
+  except Exception as ex:
+    logger.error(f'Failed to retrieve the list of traits: {ex}')
+    abort(500)
+
   return render_template('tools/phenotype_database/phenotypedb.html', **{
     # Page info
     "title": 'Phenotype Database and Analysis',
     "tool_alt_parent_breadcrumb": { "title": "Tools", "url": url_for('tools.tools') },
+
+    # Data
+    'traits': traits_non_bulk,
   })
 
+@phenotype_database_bp.route('/traits-zhang')
+@cache.memoize(60*60)
+@compress.compressed()
+def get_zhang_traits_json():
+  """
+    Phenotype Database table (bulk)
+    Fetch table content by request for each page and render the data for Datatables()
+  """
+  try:
+    # get parameters for query
+    draw = request.args.get('draw', type=int)
+    start = request.args.get('start', type=int)
+    length = request.args.get('length', type=int)
+    search_value = request.args.get('search[value]', '').lower()
+
+    query = PhenotypeMetadata.query
+    if search_value:
+      query = query.filter(
+        or_(
+          func.lower(PhenotypeMetadata.trait_name_caendr.like(f"%{search_value}%")),
+          func.lower(PhenotypeMetadata.trait_name_user.like(f"%{search_value}%")),
+          func.lower(PhenotypeMetadata.description_short.like(f"%{search_value}%")),
+          func.lower(PhenotypeMetadata.description_long.like(f"%{search_value}%")),
+          func.lower(PhenotypeMetadata.source_lab.like(f"%{search_value}%")),
+          func.lower(PhenotypeMetadata.institution.like(f"%{search_value}%")),
+          func.lower(PhenotypeMetadata.submitted_by.like(f"%{search_value}%")),
+        )
+      )
+
+    # Query PhenotypeMetadata (include phenotype values for each trait)
+    data = query.offset(start).limit(length).from_self().\
+      join(PhenotypeMetadata.phenotype_values).all()
+    
+    json_data = convert_joined_query_tojson(data)
+    
+    total_records = query.count()
+
+    response_data = {
+        "draw": draw,
+        "recordsTotal": total_records,
```

Comment:
if possible, we should use snake-case for json payloads. (eg: `"total_records": total_records, `)